lib/quintype/api/story.rb

```diff
@@ -76,15 +76,28 @@ def cards
       @cards = story['cards'] || []
     end
 
-    def to_h(config={})
+    def to_h(config={}, page_type=nil)
       hash = story.merge({
         'url' => URL.story(story),
-        'canonical_url' => URL.story_canonical(config['root_url'], story),
         'time_in_minutes' => time_in_minutes,
         'tags' => add_urls_to_tags
       })
       if config.present?
-        hash.merge!({ 'sections' => add_display_names_to_sections(config) })
+        hash.merge!({
+          'sections' => add_display_names_to_sections(config),
+          'canonical_url' => URL.story_canonical(config['root_url'], story),
+        })
+      end
+      if story['alternative'].present? && page_type
```

Comment:
put this into a separate function, called display_headline(page_type = nil). Don't put so much logic into an anonymous function like to_h.

Apply the rule of 7, it's more obvious in ruby. If any function is 7 lines long, it's too damn long. In fact, I'd put alternates into a separate class all together Alternates.new(self).headline